auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealmBuilder.java

```diff
@@ -154,6 +193,34 @@ public FileSystemSecurityRealm build() {
         if (hashCharset == null) {
             hashCharset = StandardCharsets.UTF_8;
         }
-        return new FileSystemSecurityRealm(root, nameRewriter, levels, encoded, hashEncoding, hashCharset, providers, secretKey);
+        if (providers == null) {
+            providers = INSTALLED_PROVIDERS;
+        }
+
+        if (privateKey == null ^ publicKey == null) {
+            throw ElytronMessages.log.invalidKeyPairArgument(root.toString());
+        }
+
+        // Validate KeyPair
+        if (privateKey != null) {
+            try {
+                byte[] challenge = new byte[10000];
+                ThreadLocalRandom.current().nextBytes(challenge);
+                Signature sig = Signature.getInstance("SHA256withRSA");
+                sig.initSign(privateKey);
+                sig.update(challenge);
+                byte[] signature = sig.sign();
+                sig.initVerify(publicKey);
+                sig.update(challenge);
+                boolean keyPairMatches = sig.verify(signature);
```

Comment:
I don't think this block is needed. We should assume that if a key pair has been provided, then it is valid.